website/client/src/pages/private-messages.vue

```diff
@@ -925,7 +924,12 @@ export default {
       }).then(response => {
         const newMessage = response.data.data.message;
         const messageToReset = messages[messages.length - 1];
-        messageToReset.id = newMessage.id; // just set the id, all other infos already set
+
+        // Update newest message with id and text if there's an @ mention
```

Comment:
@kevechen I think there's actually a bug at line 933 with `Object.assign(messages[messages.length - 1], messageToReset);` since it's assigning an object to itself (`messageToReset` is `messages[messages.length - 1];`) so I think the easiest way to fix this would be to change the code to something like this:

```
}).then(response => {
  const newMessage = response.data.data.message;
  this.selectedConversation.lastMessageText = newMessage.text;
  Object.assign(messages[messages.length - 1], newMessage);
  this.updateConversationsCounter += 1;
});
```

Reply:
@paglias I've modified it with your suggestion, but the message does not show up correctly. It might be because the response's message has a different data structure than the message we're trying to set (results below -- this is a conversation from testing1 to testing2).

*With those changes:*
<img width="649" alt="Screen Shot 2020-12-13 at 1 23 05 PM" src="https://user-images.githubusercontent.com/33815349/102020302-64053500-3d46-11eb-9d2e-46a1b49df463.png">

*After reloading the page:*
<img width="638" alt="Screen Shot 2020-12-13 at 1 23 17 PM" src="https://user-images.githubusercontent.com/33815349/102020308-68315280-3d46-11eb-8236-36c2691363bf.png">

I think the reason we had messageToReset was to get the original structure of a single message first, then set the changes, and copy it over. We may not even need that variable, so maybe this is better:
```
}).then(response => {
        const newMessage = response.data.data.message;
        messages[messages.length - 1].id = newMessage.id;
        messages[messages.length - 1].text = newMessage.text;
        this.selectedConversation.lastMessageText = newMessage.text;
        this.updateConversationsCounter += 1;
});
```

Reply:
Hmm I think that could be because Vue by default doesn't recognize new properties. Can you fix the commits or open a new PR? If we change the object completely it should work, so something like `messages[messages.length - 1] = newMessage`

Reply:
@paglias I made a new PR #12883 with the suggested changes, so I'll go ahead and close this one!